src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs

```diff
@@ -49,53 +60,64 @@ public static bool FileExists(string fullPath)
         /// <param name="returnErrorOnNotFound">Return the error code for not found errors?</param>
         internal static int FillAttributeInfo(string? path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool returnErrorOnNotFound)
         {
-            int errorCode = Interop.Errors.ERROR_SUCCESS;
-
             // Neither GetFileAttributes or FindFirstFile like trailing separators
             path = PathInternal.TrimEndingDirectorySeparator(path);
 
             using (DisableMediaInsertionPrompt.Create())
             {
-                if (!Interop.Kernel32.GetFileAttributesEx(path, Interop.Kernel32.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, ref data))
+                return FillAttributeInfoSlim(path, ref data, returnErrorOnNotFound);
+            }
+        }
+
+        /// <summary>
+        /// Same as <see cref="FillAttributeInfo"/> without separator-trimming and media-insertion blocking
+        /// </summary>
+        /// <param name="path">The file path from which the file attribute information will be filled.</param>
+        /// <param name="data">A struct that will contain the attribute information.</param>
+        /// <param name="returnErrorOnNotFound">Return the error code for not found errors?</param>
+        internal static int FillAttributeInfoSlim(string? path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool returnErrorOnNotFound)
+        {
+            int errorCode = Interop.Errors.ERROR_SUCCESS;
+            string? prefixedString = PathInternal.EnsureExtendedPrefixIfNeeded(path);
```

Comment:
The FindFirstFilePrefixed path here is the rare case, right? (<1%) Where the file is locked.

Given that, I don't think you're really saving anything by ensuring you only call EnsureExtendedPrefixIfNeeded once. I would reverse this and leave GetFileAttributesExPrivate private. That avoids creating a way for future code to accidentally call GetFileAttributesEx without prefixing. And it leaves this code a bit simpler.

Reply:
This was marked resolved but I don't see the feedback addressed.  Was it commented on somewhere else?